core/DataAccess/LogAggregator.php

```diff
@@ -1083,6 +1083,139 @@ public function queryConversionsByDimension($dimensions = array(), $where = fals
+     */
+    public function queryConversionsByPageView(string $linkField, $rankingQueryLimit = 0)
+    {
+        $dbSettings = new \Piwik\Db\Settings();
+        $tablePrefix = $dbSettings->getTablePrefix();
+        $subQuery = sprintf("
+            MAX((SELECT COUNT(am.idaction)
+             FROM %slog_conversion cam
+             LEFT JOIN %slog_link_visit_action vam ON vam.idvisit = cam.idvisit
+             LEFT JOIN %slog_action am ON am.idaction = vam.idaction_url
+             WHERE cam.idgoal = log_conversion.idgoal AND vam.idvisit = log_link_visit_action.idvisit
+             AND vam.idaction_url IS NOT NULL AND am.type = 1
+             AND vam.server_time <= log_conversion.server_time
+             GROUP BY cam.idgoal, cam.idvisit)) AS `9`
+        ", $tablePrefix, $tablePrefix, $tablePrefix);
+
+        // Get unique pages visited before the goal conversion, one row per goal / visit / page combination
+        $query = $this->generateQuery(
+        // SELECT ...
+            implode(
+                ', ',
+                array(
+                    "log_conversion.idgoal AS idgoal",
+                    "log_link_visit_action.".$linkField." AS idaction",
+                    "log_action.type",
+                    "count(*) AS `10`",
+                    "count(distinct log_conversion.idvisit) AS `3`",
+                    sprintf('%s AS `%d`', self::getSqlRevenue('SUM(log_conversion.revenue)'), 2),
+                    sprintf('%s AS `%d`', self::getSqlRevenue('SUM(log_conversion.revenue_subtotal)'), 4),
+                    sprintf('%s AS `%d`', self::getSqlRevenue('SUM(log_conversion.revenue_tax)'), 5),
+                    sprintf('%s AS `%d`', self::getSqlRevenue('SUM(log_conversion.revenue_shipping)'), 6),
+                    sprintf('%s AS `%d`', self::getSqlRevenue('SUM(log_conversion.revenue_discount)'), 7),
+                    "SUM(log_conversion.items) AS `8`",
+                    $subQuery,
+                    "1 AS `12`",
+                    "count(*) AS `1`",
+                )
+            ),
+            // FROM...
+            array(
+                self::LOG_CONVERSION_TABLE,
+                array(
+                    "table" => "log_link_visit_action",
+                    "joinOn" => "log_link_visit_action.idvisit = log_conversion.idvisit AND log_link_visit_action.server_time <= log_conversion.server_time"
+                ),
+                array(
+                    "table" => "log_action",
+                    "joinOn" => "log_action.idaction = log_link_visit_action.".$linkField
+                )
+            ),
+            // WHERE ... AND ...
+            implode(
+                ' AND ',
+                array(
+                    'log_conversion.server_time >= ?',
+                    'log_conversion.server_time <= ?',
+                    'log_conversion.idsite IN ('.Common::getSqlStringFieldsArray($this->sites).')',
+                    'log_conversion.idgoal > 0',
```

Comment:
Shouldn't that be `>=0` so the metrics are also calculated for ecommerce?
```suggestion
                    'log_conversion.idgoal >= 0',
```